cirq-google/cirq_google/api/v2/run_context.proto

```diff
@@ -127,15 +127,37 @@ message DeviceParametersDiff {
     // as index into the strs repeated field.
     int32 name = 2;
   }
+
+  // Param value whose type is not among proto field types supported by
+  // ArgValue. In other words, it is the responsibility of the client codes
+  // to enforce the validity and type consistency of this param value.
+  message GenericValue {
+    // Type description of the value representation. Eg. if the following value
+    // bytes field is a JSON string, type_descriptor can be its JSON namespace;
+    // or if the following value field is a protobuf serialization, type_descriptor
+    // can be its protobuf definition URL.
+    string type_descriptor = 1;
+
+    // The value in client's encoding.
+    bytes value = 2;
+  }
+
   message Param {
     // the resource group hosting this parameter key, as index into groups
     // repeated field.
     int32 resource_group = 1;
     // name of this param, as index into the strs repeated field.
     int32 name = 2;
-    // this param's new value, as message ArgValue to allow variant types,
-    // including bool, string, double, float and arrays.
-    ArgValue value = 3;
+
+    oneof param_val {
+      // this param's new value, as message ArgValue to allow types of bool,
+      // string, double, float and arrays.
+      ArgValue value = 3;
+
+      // this param's new value, and its type is not among the variants supported
+      // by ArgValue.
+      GenericValue generic_value = 5;
```

Comment:
Is it intentional to use 5 instead of 4?

Reply:
yes, 4 is already taken by
```
repeated string strs = 4;
```